### PR TITLE
[EventHubs] Auth consolidation

### PR DESCRIPTION
**Pull request by annatisch**
Fix for issue #8603 

- Removed legacy auth code
- Removed explicit SAS token auth
- Updated SharedCredentialAuth to align to TokenCredential protocol

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhubs/azure/eventhub/_client_base.py [hunk 1]
```diff
@@ -158,39 +166,29 @@ def _create_auth(self):
...
+        if token_type == b"servicebus.windows.net:sastoken":
+            auth = authentication.JWTTokenAuth(
+                self._auth_uri,
+                self._auth_uri,
+                functools.partial(self._credential.get_token, self._auth_uri),
+                token_type=token_type,
+                timeout=self._config.auth_timeout,
+                http_proxy=self._config.http_proxy,
+                transport_type=self._config.transport_type)
+            auth.update_token()
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/_client_base.py]: I think this one should work but do we need to call `update_token()` here?
it's already being called in the uamqp client_ready() right?

Comment (annatisch) [reply sdk/eventhub/azure-eventhubs/azure/eventhub/_client_base.py]: Unfortunately in uamqp we only call update_token if the type is jwt, which it isn't in this case:
https://github.com/Azure/azure-uamqp-python/blob/master/uamqp/authentication/cbs_auth.py#L66

**Pull request merged**